### PR TITLE
Fix inconsistencies for the tree view interface

### LINKED ISSUE #15676: Visual inconsistency for tree-view
### Describe the Bug

The tree-view interface has some visual quirks as compared to the other relational interfaces, mainly:

- It uses shadows instead of the bordered row style
- Open/close arrows point the wrong way (closed should be 👉🏻, open should be 👇🏻)
- There's no notice when there aren't any items yet

### To Reproduce

Add a tree-view interface with some content,  note the visual inconsistencies:

CleanShot 2022-09-20 at 16 32 03@2x

### Errors Shown

n/a

### What version of Directus are you using?

v9.17.4

### What version of Node.js are you using?

18

### What database are you using?

PG13

### What browser are you using?

Safari

### How are you deploying Directus?

Locally

### PR DESCRIPTION
**Pull request by ComfortablyCoding**
## Scope

What's changed:

- Fixes some minor inconsistencies for tree view interface as mentioned in the issue.
- Replaces usage of `ul/li` with `v-list/v-list-item`.
- Adds a loading skeleton for initial data load.

#### Before

Screenshot 2024-01-08 at 10 55 19 PM

#### After

Screenshot 2024-01-19 at 3 59 56 PM

#### Demo

https://github.com/directus/directus/assets/44623501/93221306-e549-4488-b2ba-d4a784b04692

## Potential Risks / Drawbacks

- Visual glitches with tree interface
- Their is a little shuffle when data loads to fast and the parent has not children. This is due to the added skeleton. It should be fine for the short term, we can look in to adding some sort of debounced loader if any complaint. Observe the difference between loading 2 & 3 vs 4 in the video below.

https://github.com/directus/directus/assets/44623501/c45be028-890a-4f09-a1b2-ffb0cc41ba67

## Review Notes / Questions

- "No Items" notice is at the root level only.

---

Fixes #15676

### CODE REVIEW

File: .changeset/rotten-goats-clap.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@directus/app': patch
+---
+
+Fixed visual insistencies for the tree view interface. Replaced `ul/li` usage with `v-list/v-list-item`. Added skeleton loader
```
Comment (paescuj) [inline .changeset/rotten-goats-clap.md:5]:
```suggestion
Fixed visual inconsistencies in the tree view interface
```

**Review by paescuj — DISMISSED** [review]
### Indent

The "No items" notice on root level shouldn't be indented:



Same goes for the loader skeleton on root level (you will notice it has the wrong indent for a split second):

https://github.com/directus/directus/assets/5363448/85fabafc-375b-4fcf-9282-2f0ff64ddb0d

### Notice & Loader Heights

The loader skeleton and "No items" notice are smaller than the items itself, I guess because items have additional border heights.
Aligning the heights should help preventing a jumping effect.

<table>
<tr><th>Item</th><th>Loader</th>
</tr><td></td><td></td></table>

### Dragging

- It feels somewhat harder now to drop the item at a nested level (smaller drop area than before?)
- A scrollbar appears now at the bottom while dragging
- Ideally, dragging to an empty, nested level would replace (hide) the "No items" notice -  not sure how difficult this is to implement, though 

#### Before

https://github.com/directus/directus/assets/5363448/726406b8-8ca8-498a-b3ba-69566acf8211

#### Now

https://github.com/directus/directus/assets/5363448/4ce4ac5b-a4e8-45ae-b6c3-4bab7c4af0ac

**Comment by ComfortablyCoding** [comment]
Indentation and notice/loader height issues should be resolved

**Comment by paescuj** [comment]
Awesome 🚀 

> - Ideally, dragging to an empty, nested level would replace (hide) the "No items" notice - not sure how difficult this is to implement, though
>     * I cannot reproduce this, for me on safari/FF the no items notice disappears if an item is dropped into the parent

Oh, I meant when dragging over, not only after dropping.

**Comment by ComfortablyCoding** [comment]
> Oh, I meant when dragging over, not only after dropping.

When we drag over an item it does not hide, doing it only for No Items might be somewhat strange.

**Comment by LasseRosenow** [comment]
I find the whole `no entries` card not so convenient.
I think it would be better not just make entries without children not expandable and hide the arrow.

This way it would also immediately be visible to the eye if a card has children or not which I think is very important.

**Comment by paescuj** [comment]
> I find the whole `no items` card not so convenient. I think it would be better not just make entries without children not expandable and hide the arrow.
> 
> This way it would also immediately be visible to the eye if a card has children or not which I think is very important.

Hmm, yeah I like that idea 👍

**Comment by ComfortablyCoding** [comment]
That would prevent dragging/dropping an item as a child of that item.

**Comment by paescuj** [comment]
> I believe that would prevent dragging/dropping an item as a child of that item. This would be a change in functionality as currently you can do so.

Wouldn't that be the same as it used to be before that PR, just hiding the arrow if there are no nested items?

**Comment by ComfortablyCoding** [comment]
If the arrow is hidden there will not be a drop zone for adding a child. Currently if you expand the empty item you can add a child to it.

#### Demo from latest release

https://github.com/directus/directus/assets/44623501/dd4a9355-13cf-4880-9c2e-8a7bd19329ba

**Comment by ComfortablyCoding** [comment]
If we remove the "No Items" on non root levels we have the following options for items with no children:
- We keep current functionality with empty expand.
- We keep current functionality with empty expand and add an indicator icon for no child item.
- We remove the expand functionality and allow a smaller drag/drop zone (e.g. 70% width of parent). This way a child can be set without it being expanded and also allow easy parse for empty child items. This is inline with other tree view layout paradigms. This should be possible but I have not attempted it.

**Comment by paescuj** [comment]
Thanks @ComfortablyCoding for pointing out the possibilities!

Personally, I tend towards the third option. Main reason for this is that it seems counter-intuitive to me that you expand an item only to see the "No items" notice.

@rijkvanzanten Would love your input here 🎨

**Comment by ComfortablyCoding** [comment]
Per rijk, option 3 is the way to go.

**Comment by ComfortablyCoding** [comment]
@paescuj Due to the issues that popup when attempting to implement option 3, I am leaving it as option 1 for now. We can revisit it if/when there is enough demand for it.

I have updated the PR description with the latest functionality/risks/notes

**Review by paescuj — COMMENTED** [review]
The buttons on the right are collapsing if the item text is too wide. Could you take a look at this?

**Review by paescuj — APPROVED** [review]
❤️

**Pull request merged**